subprojects/language-java/src/integTest/groovy/org/gradle/api/tasks/compile/AggregatingIncrementalAnnotationProcessingIntegrationTest.groovy

```diff
@@ -204,4 +208,11 @@ class AggregatingIncrementalAnnotationProcessingIntegrationTest extends Abstract
         expect:
         succeeds "compileJava"
     }
+
+    private boolean serviceRegistryReferences(String... services) {
```

Comment:
I think this method should also test `ServiceRegistry.class` file, not only `ServiceRegistry.java`.